### PR TITLE
🔧 Deploy website on Netlify

### PR DESCRIPTION
**Pull request by dubzzz**
**_Category:_**

- [ ] ✨ Introduce new features
- [ ] 📝 Add or update documentation
- [ ] ✅ Add or update tests
- [ ] 🐛 Fix a bug
- [ ] 🏷️ Add or update types
- [ ] ⚡️ Improve performance
- [ ] _Other(s):_ ...
  
  

**_Potential impacts:_**

- [ ] Generated values
- [ ] Shrink values
- [ ] Performance
- [ ] Typings
- [ ] _Other(s):_ ...

### CODE REVIEW

File: .github/workflows/build-status.yml [hunk 1]
```diff
@@ -517,6 +524,34 @@ jobs:
...
+      - name: Download documentation artifacts
+        uses: actions/download-artifact@87c55149d96e628cc2ef7e6fc2aab372015aec85 # v4.1.3
+        with:
+          name: website
+          path: artifacts/
+      - name: Deploy to Netlify
+        run: |
+          prod_flag=""
+          if [ "$BRANCH_NAME" = "main" ]; then prod_flag="--prod"; fi
+          cd website/ && yarn netlify-deploy --dir ../artifacts --site ${{secrets.NETLIFY_SITE_ID}} --auth ${{secrets.NETLIFY_AUTH_TOKEN}} $prod_flag
```
Comment (dubzzz) [inline .github/workflows/build-status.yml:554]:
```suggestion
          cd website/ && ls -alhR ../artifacts && cwd && yarn netlify-deploy --dir ../artifacts --site ${{secrets.NETLIFY_SITE_ID}} --auth ${{secrets.NETLIFY_AUTH_TOKEN}} $prod_flag
```

File: .github/workflows/build-status.yml [hunk 2]
```diff
@@ -517,6 +524,34 @@ jobs:
...
+      - name: Download documentation artifacts
+        uses: actions/download-artifact@87c55149d96e628cc2ef7e6fc2aab372015aec85 # v4.1.3
+        with:
+          name: website
+          path: artifacts/
+      - name: Deploy to Netlify
+        run: |
+          prod_flag=""
+          if [ "$BRANCH_NAME" = "main" ]; then prod_flag="--prod"; fi
+          cd website/ && ls -alhR ../artifacts && cwd && yarn netlify-deploy --dir ../artifacts --site ${{secrets.NETLIFY_SITE_ID}} --auth ${{secrets.NETLIFY_AUTH_TOKEN}} $prod_flag
```
Comment (dubzzz) [inline .github/workflows/build-status.yml:554]:
```suggestion
          cd website/ && ls -alhR ../artifacts && pwd && yarn netlify-deploy --dir ../artifacts --site ${{secrets.NETLIFY_SITE_ID}} --auth ${{secrets.NETLIFY_AUTH_TOKEN}} $prod_flag
```

File: website/package.json [hunk 3]
```diff
@@ -37,6 +38,7 @@
     "fast-check": "workspace:*",
     "jest": "^29.7.0",
     "jimp": "^0.22.12",
+    "netlify-cli": "^17.17.1",
```
Comment (dubzzz) [inline website/package.json:41]: Should be replaced by https://docs.netlify.com/api/get-started/#zip-file-method

File: website/docusaurus.config.ts [hunk 4]
```diff
@@ -46,10 +46,6 @@ const config: Config = {
         },
         blog: { showReadingTime: true },
         theme: { customCss: require.resolve('./src/css/custom.css') },
```
Comment (dubzzz) [inline website/docusaurus.config.ts:48]:
```suggestion
        theme: { customCss: require.resolve('./src/css/custom.css') },
        gtag: {
          trackingID: 'G-PHXW2XCMCL',
          anonymizeIP: true,
        },
```

File: website/package.json [hunk 5]
```diff
@@ -7,6 +7,7 @@
     "build": "cp ../.all-contributorsrc src/components/HomepageContributors/all-contributors.json && node prebuild/optimize-images.mjs && docusaurus build",
     "swizzle": "docusaurus swizzle",
     "deploy": "docusaurus deploy",
+    "netlify-deploy": "netlify deploy",
```
Comment (dubzzz) [inline website/package.json:10]:
```suggestion
```

File: .gitignore [hunk 6]
```diff
@@ -22,3 +22,6 @@ v8.out
 !.yarn/releases
 !.yarn/sdks
 !.yarn/versions
+
+# Local Netlify folder
+.netlify
```
Comment (dubzzz) [inline .gitignore:27]:
```suggestion
!.yarn/versions
```

**Comment by gruhn** [comment]
Hey man, only a few days ago I migrated everything I had away from Netlify after reading this story:

https://www.reddit.com/r/webdev/comments/1b14bty/netlify_just_sent_me_a_104k_bill_for_a_simple/

The actual risk is probably low. But still very scary situation :S As others have suggested, a viable alternative seems to be [Cloudflare Pages](https://pages.cloudflare.com/)

**Pull request merged**